projects/packages/sync/src/class-queue.php

```diff
@@ -367,34 +302,37 @@ public function checkout_with_memory_limit( $max_memory, $max_buffer_size = 500
 				break;
 			}
 
-			$max_item_id = $item_with_size->id;
+			$item_ids_to_fetch[] = $item_with_size->id;
 		}
 
-		$query = $wpdb->prepare(
-			"SELECT option_name AS id, option_value AS value FROM $wpdb->options WHERE option_name >= %s and option_name <= %s ORDER BY option_name ASC",
-			$min_item_id,
-			$max_item_id
-		);
+		$current_items = $this->queue_storage->fetch_items_by_ids( $item_ids_to_fetch );
```

Comment:
I do have some performance concerns about replacing the `WHERE option_name >= %s and option_name <= %s` with `WHERE IN` for at least 500 items. Any reasons we didn't keep the original logic here?

Reply:
There should not be an issue, as we're calculating memory limitations beforehand.

In this case we first calculate the size and items that we want to include in the batch and then fetch only the items we need.

If we see issues, we can always `array_chunk` the IDs and fetch them in smaller batches.